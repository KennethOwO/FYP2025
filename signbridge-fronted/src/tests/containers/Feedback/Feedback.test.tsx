import { create } from "react-test-renderer";
import {
  render,
  fireEvent,
  cleanup,
  screen,
  waitFor,
} from "@testing-library/react";
import Feedback from "../,./../../../containers/Feedback/Feedback";
import { userEvent } from "@testing-library/user-event";

const mockedUsedNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));

jest.mock("react-i18next", () => ({
  useTranslation: () => ({
    t: (key: string) => key, // Mock the translation function to return the key
    i18n: {
      changeLanguage: jest.fn(), // Mock the changeLanguage function
    },
  }),
}));
describe("Test Feedback", () => {
  afterEach(() => {
    // Clean up the DOM
    cleanup();
    // Reset mocks to their initial state
    jest.resetAllMocks();
  });
  it("should render correctly", () => {
    const tree = create(<Feedback />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should be able to submit", async () => {
    const { asFragment } = render(<Feedback />);
    // Fill in the form fields
    userEvent.type(screen.getByTestId("test_firstName"), "John");
    userEvent.type(screen.getByTestId("test_lastName"), "Doe");
    userEvent.type(screen.getByTestId("test_age"), "5");
    userEvent.click(screen.getByTestId("test_male"));
    userEvent.type(screen.getByTestId("test_email"), "john@BiLogoGmail.com");
    userEvent.type(screen.getByTestId("test_race"), "Asian");

    // Click the submit button
    fireEvent.click(screen.getByTestId("test_submit"));
    expect(asFragment).toMatchSnapshot();
  });
});
