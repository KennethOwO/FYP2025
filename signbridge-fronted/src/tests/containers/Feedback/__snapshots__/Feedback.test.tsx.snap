// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Feedback should be able to submit 1`] = `[Function]`;

exports[`Test Feedback should render correctly 1`] = `
[
  <div>
    <h1>
      feedback_us
    </h1>
    <p>
      feedback_subheading
    </p>
  </div>,
  <div>
    <div>
      <form
        id="feedbackForm"
        onSubmit={[Function]}
      >
        <legend>
          personal_details
        </legend>
        <fieldset>
          <div>
            <div>
              <label
                htmlFor="firstName"
              >
                first_name
                 
                <span
                  style={
                    {
                      "color": "red",
                    }
                  }
                >
                  *
                </span>
              </label>
              <input
                data-testid="test_firstName"
                id="firstName"
                maxLength={25}
                name="firstName"
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
            <div>
              <label
                htmlFor="lastName"
              >
                last_name
                 
                <span
                  style={
                    {
                      "color": "red",
                    }
                  }
                >
                  *
                </span>
              </label>
              <input
                data-testid="test_lastName"
                id="lastName"
                maxLength={25}
                name="lastName"
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
          <div>
            <div>
              <label>
                age
                 
                <span
                  style={
                    {
                      "color": "red",
                    }
                  }
                >
                  *
                </span>
              </label>
              <input
                data-testid="test_age"
                id="age"
                max={150}
                min={2}
                name="age"
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
            <div>
              <label>
                gender
                 
                <span
                  style={
                    {
                      "color": "red",
                    }
                  }
                >
                  *
                </span>
              </label>
              <div>
                <input
                  checked={true}
                  data-testid="test_male"
                  id="male"
                  name="gender"
                  onChange={[Function]}
                  type="radio"
                  value="male"
                />
                <label
                  htmlFor="male"
                >
                   
                  male
                </label>
                <input
                  className="female"
                  data-testid="test_female"
                  id="female"
                  name="gender"
                  onChange={[Function]}
                  type="radio"
                  value="female"
                />
                <label
                  htmlFor="female"
                >
                   
                  female
                </label>
              </div>
            </div>
          </div>
          <div>
            <div>
              <label>
                email_address
                 
                <span
                  style={
                    {
                      "color": "red",
                    }
                  }
                >
                  *
                </span>
              </label>
              <input
                data-testid="test_email"
                id="email"
                name="email"
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
            <div>
              <label>
                race
                 
                <span
                  style={
                    {
                      "color": "red",
                    }
                  }
                >
                  *
                </span>
              </label>
              <input
                data-testid="test_race"
                id="race"
                maxLength={15}
                name="race"
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
        </fieldset>
        <legend>
          ratings
        </legend>
        <fieldset>
          <div>
            <div
              className="undefined "
            >
              <label>
                feedback_category
              </label>
              <div>
                <select
                  data-testid="test_fcategory"
                  id="fcategory"
                  onChange={[Function]}
                  onClick={[Function]}
                  value="Whole Website"
                >
                  <option
                    value="Whole Website"
                  >
                    whole_website
                  </option>
                  <option
                    value="Game 1"
                  >
                    game1_guess_the_word
                  </option>
                  <option
                    value="Game 2"
                  >
                    game2_do_the_sign
                  </option>
                </select>
                <div />
              </div>
            </div>
          </div>
        </fieldset>
        <fieldset>
          <div>
            <div>
              <label>
                experience
              </label>
              <div>
                <div>
                  <label
                    htmlFor="super-happy"
                  >
                    <input
                      checked={true}
                      className="super_happy"
                      id="super-happy"
                      name="experience"
                      onChange={[Function]}
                      type="radio"
                      value={5}
                    />
                    <svg
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12,17.5C14.33,17.5 16.3,16.04 17.11,14H6.89C7.69,16.04 9.67,17.5 12,17.5M8.5,11A1.5,1.5 0 0,0 10,9.5A1.5,1.5 0 0,0 8.5,8A1.5,1.5 0 0,0 7,9.5A1.5,1.5 0 0,0 8.5,11M15.5,11A1.5,1.5 0 0,0 17,9.5A1.5,1.5 0 0,0 15.5,8A1.5,1.5 0 0,0 14,9.5A1.5,1.5 0 0,0 15.5,11M12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4A8,8 0 0,1 20,12A8,8 0 0,1 12,20M12,2C6.47,2 2,6.5 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
                      />
                    </svg>
                  </label>
                </div>
                <div>
                  <label
                    htmlFor="happy"
                  >
                    <input
                      className="happy"
                      id="happy"
                      name="experience"
                      onChange={[Function]}
                      type="radio"
                      value={4}
                    />
                    <svg
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20,12A8,8 0 0,0 12,4A8,8 0 0,0 4,12A8,8 0 0,0 12,20A8,8 0 0,0 20,12M22,12A10,10 0 0,1 12,22A10,10 0 0,1 2,12A10,10 0 0,1 12,2A10,10 0 0,1 22,12M10,9.5C10,10.3 9.3,11 8.5,11C7.7,11 7,10.3 7,9.5C7,8.7 7.7,8 8.5,8C9.3,8 10,8.7 10,9.5M17,9.5C17,10.3 16.3,11 15.5,11C14.7,11 14,10.3 14,9.5C14,8.7 14.7,8 15.5,8C16.3,8 17,8.7 17,9.5M12,17.23C10.25,17.23 8.71,16.5 7.81,15.42L9.23,14C9.68,14.72 10.75,15.23 12,15.23C13.25,15.23 14.32,14.72 14.77,14L16.19,15.42C15.29,16.5 13.75,17.23 12,17.23Z"
                      />
                    </svg>
                  </label>
                </div>
                <div>
                  <label
                    htmlFor="neutral"
                  >
                    <input
                      className="neutral"
                      id="neutral"
                      name="experience"
                      onChange={[Function]}
                      type="radio"
                      value={3}
                    />
                    <svg
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.5,11A1.5,1.5 0 0,1 7,9.5A1.5,1.5 0 0,1 8.5,8A1.5,1.5 0 0,1 10,9.5A1.5,1.5 0 0,1 8.5,11M15.5,11A1.5,1.5 0 0,1 14,9.5A1.5,1.5 0 0,1 15.5,8A1.5,1.5 0 0,1 17,9.5A1.5,1.5 0 0,1 15.5,11M12,20A8,8 0 0,0 20,12A8,8 0 0,0 12,4A8,8 0 0,0 4,12A8,8 0 0,0 12,20M12,2A10,10 0 0,1 22,12A10,10 0 0,1 12,22C6.47,22 2,17.5 2,12A10,10 0 0,1 12,2M9,14H15A1,1 0 0,1 16,15A1,1 0 0,1 15,16H9A1,1 0 0,1 8,15A1,1 0 0,1 9,14Z"
                      />
                    </svg>
                  </label>
                </div>
                <div>
                  <label
                    htmlFor="sad"
                  >
                    <input
                      className="sad"
                      id="sad"
                      name="experience"
                      onChange={[Function]}
                      type="radio"
                      value={2}
                    />
                    <svg
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20,12A8,8 0 0,0 12,4A8,8 0 0,0 4,12A8,8 0 0,0 12,20A8,8 0 0,0 20,12M22,12A10,10 0 0,1 12,22A10,10 0 0,1 2,12A10,10 0 0,1 12,2A10,10 0 0,1 22,12M15.5,8C16.3,8 17,8.7 17,9.5C17,10.3 16.3,11 15.5,11C14.7,11 14,10.3 14,9.5C14,8.7 14.7,8 15.5,8M10,9.5C10,10.3 9.3,11 8.5,11C7.7,11 7,10.3 7,9.5C7,8.7 7.7,8 8.5,8C9.3,8 10,8.7 10,9.5M12,14C13.75,14 15.29,14.72 16.19,15.81L14.77,17.23C14.32,16.5 13.25,16 12,16C10.75,16 9.68,16.5 9.23,17.23L7.81,15.81C8.71,14.72 10.25,14 12,14Z"
                      />
                    </svg>
                  </label>
                </div>
                <div>
                  <label
                    htmlFor="super-sad"
                  >
                    <input
                      className="super_sad"
                      id="super-sad"
                      name="experience"
                      onChange={[Function]}
                      type="radio"
                      value={1}
                    />
                    <svg
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12,2C6.47,2 2,6.47 2,12C2,17.53 6.47,22 12,22A10,10 0 0,0 22,12C22,6.47 17.5,2 12,2M12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4A8,8 0 0,1 20,12A8,8 0 0,1 12,20M16.18,7.76L15.12,8.82L14.06,7.76L13,8.82L14.06,9.88L13,10.94L14.06,12L15.12,10.94L16.18,12L17.24,10.94L16.18,9.88L17.24,8.82L16.18,7.76M7.82,12L8.88,10.94L9.94,12L11,10.94L9.94,9.88L11,8.82L9.94,7.76L8.88,8.82L7.82,7.76L6.76,8.82L7.82,9.88L6.76,10.94L7.82,12M12,14C9.67,14 7.69,15.46 6.89,17.5H17.11C16.31,15.46 14.33,14 12,14Z"
                      />
                    </svg>
                  </label>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div>
              <label>
                friendliness
              </label>
              <div>
                <div>
                  <label
                    htmlFor="super-happy1"
                  >
                    <input
                      checked={true}
                      className="super_happy"
                      id="super-happy1"
                      name="friendliness"
                      onChange={[Function]}
                      type="radio"
                      value={5}
                    />
                    <svg
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12,17.5C14.33,17.5 16.3,16.04 17.11,14H6.89C7.69,16.04 9.67,17.5 12,17.5M8.5,11A1.5,1.5 0 0,0 10,9.5A1.5,1.5 0 0,0 8.5,8A1.5,1.5 0 0,0 7,9.5A1.5,1.5 0 0,0 8.5,11M15.5,11A1.5,1.5 0 0,0 17,9.5A1.5,1.5 0 0,0 15.5,8A1.5,1.5 0 0,0 14,9.5A1.5,1.5 0 0,0 15.5,11M12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4A8,8 0 0,1 20,12A8,8 0 0,1 12,20M12,2C6.47,2 2,6.5 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
                      />
                    </svg>
                  </label>
                </div>
                <div>
                  <label
                    htmlFor="happy1"
                  >
                    <input
                      className="happy"
                      id="happy1"
                      name="friendliness"
                      onChange={[Function]}
                      type="radio"
                      value={4}
                    />
                    <svg
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20,12A8,8 0 0,0 12,4A8,8 0 0,0 4,12A8,8 0 0,0 12,20A8,8 0 0,0 20,12M22,12A10,10 0 0,1 12,22A10,10 0 0,1 2,12A10,10 0 0,1 12,2A10,10 0 0,1 22,12M10,9.5C10,10.3 9.3,11 8.5,11C7.7,11 7,10.3 7,9.5C7,8.7 7.7,8 8.5,8C9.3,8 10,8.7 10,9.5M17,9.5C17,10.3 16.3,11 15.5,11C14.7,11 14,10.3 14,9.5C14,8.7 14.7,8 15.5,8C16.3,8 17,8.7 17,9.5M12,17.23C10.25,17.23 8.71,16.5 7.81,15.42L9.23,14C9.68,14.72 10.75,15.23 12,15.23C13.25,15.23 14.32,14.72 14.77,14L16.19,15.42C15.29,16.5 13.75,17.23 12,17.23Z"
                      />
                    </svg>
                  </label>
                </div>
                <div>
                  <label
                    htmlFor="neutral1"
                  >
                    <input
                      className="neutral"
                      id="neutral1"
                      name="friendliness"
                      onChange={[Function]}
                      type="radio"
                      value={3}
                    />
                    <svg
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.5,11A1.5,1.5 0 0,1 7,9.5A1.5,1.5 0 0,1 8.5,8A1.5,1.5 0 0,1 10,9.5A1.5,1.5 0 0,1 8.5,11M15.5,11A1.5,1.5 0 0,1 14,9.5A1.5,1.5 0 0,1 15.5,8A1.5,1.5 0 0,1 17,9.5A1.5,1.5 0 0,1 15.5,11M12,20A8,8 0 0,0 20,12A8,8 0 0,0 12,4A8,8 0 0,0 4,12A8,8 0 0,0 12,20M12,2A10,10 0 0,1 22,12A10,10 0 0,1 12,22C6.47,22 2,17.5 2,12A10,10 0 0,1 12,2M9,14H15A1,1 0 0,1 16,15A1,1 0 0,1 15,16H9A1,1 0 0,1 8,15A1,1 0 0,1 9,14Z"
                      />
                    </svg>
                  </label>
                </div>
                <div>
                  <label
                    htmlFor="sad1"
                  >
                    <input
                      className="sad"
                      id="sad1"
                      name="friendliness"
                      onChange={[Function]}
                      type="radio"
                      value={2}
                    />
                    <svg
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20,12A8,8 0 0,0 12,4A8,8 0 0,0 4,12A8,8 0 0,0 12,20A8,8 0 0,0 20,12M22,12A10,10 0 0,1 12,22A10,10 0 0,1 2,12A10,10 0 0,1 12,2A10,10 0 0,1 22,12M15.5,8C16.3,8 17,8.7 17,9.5C17,10.3 16.3,11 15.5,11C14.7,11 14,10.3 14,9.5C14,8.7 14.7,8 15.5,8M10,9.5C10,10.3 9.3,11 8.5,11C7.7,11 7,10.3 7,9.5C7,8.7 7.7,8 8.5,8C9.3,8 10,8.7 10,9.5M12,14C13.75,14 15.29,14.72 16.19,15.81L14.77,17.23C14.32,16.5 13.25,16 12,16C10.75,16 9.68,16.5 9.23,17.23L7.81,15.81C8.71,14.72 10.25,14 12,14Z"
                      />
                    </svg>
                  </label>
                </div>
                <div>
                  <label
                    htmlFor="super-sad1"
                  >
                    <input
                      className="super_sad"
                      id="super-sad1"
                      name="friendliness"
                      onChange={[Function]}
                      type="radio"
                      value={1}
                    />
                    <svg
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12,2C6.47,2 2,6.47 2,12C2,17.53 6.47,22 12,22A10,10 0 0,0 22,12C22,6.47 17.5,2 12,2M12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4A8,8 0 0,1 20,12A8,8 0 0,1 12,20M16.18,7.76L15.12,8.82L14.06,7.76L13,8.82L14.06,9.88L13,10.94L14.06,12L15.12,10.94L16.18,12L17.24,10.94L16.18,9.88L17.24,8.82L16.18,7.76M7.82,12L8.88,10.94L9.94,12L11,10.94L9.94,9.88L11,8.82L9.94,7.76L8.88,8.82L7.82,7.76L6.76,8.82L7.82,9.88L6.76,10.94L7.82,12M12,14C9.67,14 7.69,15.46 6.89,17.5H17.11C16.31,15.46 14.33,14 12,14Z"
                      />
                    </svg>
                  </label>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div>
              <label>
                quality
              </label>
              <div>
                <div>
                  <label
                    htmlFor="super-happy2"
                  >
                    <input
                      checked={true}
                      className="super_happy"
                      id="super-happy2"
                      name="quality"
                      onChange={[Function]}
                      type="radio"
                      value={5}
                    />
                    <svg
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12,17.5C14.33,17.5 16.3,16.04 17.11,14H6.89C7.69,16.04 9.67,17.5 12,17.5M8.5,11A1.5,1.5 0 0,0 10,9.5A1.5,1.5 0 0,0 8.5,8A1.5,1.5 0 0,0 7,9.5A1.5,1.5 0 0,0 8.5,11M15.5,11A1.5,1.5 0 0,0 17,9.5A1.5,1.5 0 0,0 15.5,8A1.5,1.5 0 0,0 14,9.5A1.5,1.5 0 0,0 15.5,11M12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4A8,8 0 0,1 20,12A8,8 0 0,1 12,20M12,2C6.47,2 2,6.5 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
                      />
                    </svg>
                  </label>
                </div>
                <div>
                  <label
                    htmlFor="happy2"
                  >
                    <input
                      className="happy"
                      id="happy2"
                      name="quality"
                      onChange={[Function]}
                      type="radio"
                      value={4}
                    />
                    <svg
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20,12A8,8 0 0,0 12,4A8,8 0 0,0 4,12A8,8 0 0,0 12,20A8,8 0 0,0 20,12M22,12A10,10 0 0,1 12,22A10,10 0 0,1 2,12A10,10 0 0,1 12,2A10,10 0 0,1 22,12M10,9.5C10,10.3 9.3,11 8.5,11C7.7,11 7,10.3 7,9.5C7,8.7 7.7,8 8.5,8C9.3,8 10,8.7 10,9.5M17,9.5C17,10.3 16.3,11 15.5,11C14.7,11 14,10.3 14,9.5C14,8.7 14.7,8 15.5,8C16.3,8 17,8.7 17,9.5M12,17.23C10.25,17.23 8.71,16.5 7.81,15.42L9.23,14C9.68,14.72 10.75,15.23 12,15.23C13.25,15.23 14.32,14.72 14.77,14L16.19,15.42C15.29,16.5 13.75,17.23 12,17.23Z"
                      />
                    </svg>
                  </label>
                </div>
                <div>
                  <label
                    htmlFor="neutral2"
                  >
                    <input
                      className="neutral"
                      id="neutral2"
                      name="quality"
                      onChange={[Function]}
                      type="radio"
                      value={3}
                    />
                    <svg
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.5,11A1.5,1.5 0 0,1 7,9.5A1.5,1.5 0 0,1 8.5,8A1.5,1.5 0 0,1 10,9.5A1.5,1.5 0 0,1 8.5,11M15.5,11A1.5,1.5 0 0,1 14,9.5A1.5,1.5 0 0,1 15.5,8A1.5,1.5 0 0,1 17,9.5A1.5,1.5 0 0,1 15.5,11M12,20A8,8 0 0,0 20,12A8,8 0 0,0 12,4A8,8 0 0,0 4,12A8,8 0 0,0 12,20M12,2A10,10 0 0,1 22,12A10,10 0 0,1 12,22C6.47,22 2,17.5 2,12A10,10 0 0,1 12,2M9,14H15A1,1 0 0,1 16,15A1,1 0 0,1 15,16H9A1,1 0 0,1 8,15A1,1 0 0,1 9,14Z"
                      />
                    </svg>
                  </label>
                </div>
                <div>
                  <label
                    htmlFor="sad2"
                  >
                    <input
                      className="sad"
                      id="sad2"
                      name="quality"
                      onChange={[Function]}
                      type="radio"
                      value={2}
                    />
                    <svg
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20,12A8,8 0 0,0 12,4A8,8 0 0,0 4,12A8,8 0 0,0 12,20A8,8 0 0,0 20,12M22,12A10,10 0 0,1 12,22A10,10 0 0,1 2,12A10,10 0 0,1 12,2A10,10 0 0,1 22,12M15.5,8C16.3,8 17,8.7 17,9.5C17,10.3 16.3,11 15.5,11C14.7,11 14,10.3 14,9.5C14,8.7 14.7,8 15.5,8M10,9.5C10,10.3 9.3,11 8.5,11C7.7,11 7,10.3 7,9.5C7,8.7 7.7,8 8.5,8C9.3,8 10,8.7 10,9.5M12,14C13.75,14 15.29,14.72 16.19,15.81L14.77,17.23C14.32,16.5 13.25,16 12,16C10.75,16 9.68,16.5 9.23,17.23L7.81,15.81C8.71,14.72 10.25,14 12,14Z"
                      />
                    </svg>
                  </label>
                </div>
                <div>
                  <label
                    htmlFor="super-sad2"
                  >
                    <input
                      className="super_sad"
                      id="super-sad2"
                      name="quality"
                      onChange={[Function]}
                      type="radio"
                      value={1}
                    />
                    <svg
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12,2C6.47,2 2,6.47 2,12C2,17.53 6.47,22 12,22A10,10 0 0,0 22,12C22,6.47 17.5,2 12,2M12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4A8,8 0 0,1 20,12A8,8 0 0,1 12,20M16.18,7.76L15.12,8.82L14.06,7.76L13,8.82L14.06,9.88L13,10.94L14.06,12L15.12,10.94L16.18,12L17.24,10.94L16.18,9.88L17.24,8.82L16.18,7.76M7.82,12L8.88,10.94L9.94,12L11,10.94L9.94,9.88L11,8.82L9.94,7.76L8.88,8.82L7.82,7.76L6.76,8.82L7.82,9.88L6.76,10.94L7.82,12M12,14C9.67,14 7.69,15.46 6.89,17.5H17.11C16.31,15.46 14.33,14 12,14Z"
                      />
                    </svg>
                  </label>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div>
              <label>
                recommended
              </label>
              <div>
                <div>
                  <label
                    htmlFor="super-happy3"
                  >
                    <input
                      checked={true}
                      className="super_happy"
                      id="super-happy3"
                      name="recommended"
                      onChange={[Function]}
                      type="radio"
                      value={5}
                    />
                    <svg
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12,17.5C14.33,17.5 16.3,16.04 17.11,14H6.89C7.69,16.04 9.67,17.5 12,17.5M8.5,11A1.5,1.5 0 0,0 10,9.5A1.5,1.5 0 0,0 8.5,8A1.5,1.5 0 0,0 7,9.5A1.5,1.5 0 0,0 8.5,11M15.5,11A1.5,1.5 0 0,0 17,9.5A1.5,1.5 0 0,0 15.5,8A1.5,1.5 0 0,0 14,9.5A1.5,1.5 0 0,0 15.5,11M12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4A8,8 0 0,1 20,12A8,8 0 0,1 12,20M12,2C6.47,2 2,6.5 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
                      />
                    </svg>
                  </label>
                </div>
                <div>
                  <label
                    htmlFor="happy3"
                  >
                    <input
                      className="happy"
                      id="happy3"
                      name="recommended"
                      onChange={[Function]}
                      type="radio"
                      value={4}
                    />
                    <svg
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20,12A8,8 0 0,0 12,4A8,8 0 0,0 4,12A8,8 0 0,0 12,20A8,8 0 0,0 20,12M22,12A10,10 0 0,1 12,22A10,10 0 0,1 2,12A10,10 0 0,1 12,2A10,10 0 0,1 22,12M10,9.5C10,10.3 9.3,11 8.5,11C7.7,11 7,10.3 7,9.5C7,8.7 7.7,8 8.5,8C9.3,8 10,8.7 10,9.5M17,9.5C17,10.3 16.3,11 15.5,11C14.7,11 14,10.3 14,9.5C14,8.7 14.7,8 15.5,8C16.3,8 17,8.7 17,9.5M12,17.23C10.25,17.23 8.71,16.5 7.81,15.42L9.23,14C9.68,14.72 10.75,15.23 12,15.23C13.25,15.23 14.32,14.72 14.77,14L16.19,15.42C15.29,16.5 13.75,17.23 12,17.23Z"
                      />
                    </svg>
                  </label>
                </div>
                <div>
                  <label
                    htmlFor="neutral3"
                  >
                    <input
                      className="neutral"
                      id="neutral3"
                      name="recommended"
                      onChange={[Function]}
                      type="radio"
                      value={3}
                    />
                    <svg
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.5,11A1.5,1.5 0 0,1 7,9.5A1.5,1.5 0 0,1 8.5,8A1.5,1.5 0 0,1 10,9.5A1.5,1.5 0 0,1 8.5,11M15.5,11A1.5,1.5 0 0,1 14,9.5A1.5,1.5 0 0,1 15.5,8A1.5,1.5 0 0,1 17,9.5A1.5,1.5 0 0,1 15.5,11M12,20A8,8 0 0,0 20,12A8,8 0 0,0 12,4A8,8 0 0,0 4,12A8,8 0 0,0 12,20M12,2A10,10 0 0,1 22,12A10,10 0 0,1 12,22C6.47,22 2,17.5 2,12A10,10 0 0,1 12,2M9,14H15A1,1 0 0,1 16,15A1,1 0 0,1 15,16H9A1,1 0 0,1 8,15A1,1 0 0,1 9,14Z"
                      />
                    </svg>
                  </label>
                </div>
                <div>
                  <label
                    htmlFor="sad3"
                  >
                    <input
                      className="sad"
                      id="sad3"
                      name="recommended"
                      onChange={[Function]}
                      type="radio"
                      value={2}
                    />
                    <svg
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20,12A8,8 0 0,0 12,4A8,8 0 0,0 4,12A8,8 0 0,0 12,20A8,8 0 0,0 20,12M22,12A10,10 0 0,1 12,22A10,10 0 0,1 2,12A10,10 0 0,1 12,2A10,10 0 0,1 22,12M15.5,8C16.3,8 17,8.7 17,9.5C17,10.3 16.3,11 15.5,11C14.7,11 14,10.3 14,9.5C14,8.7 14.7,8 15.5,8M10,9.5C10,10.3 9.3,11 8.5,11C7.7,11 7,10.3 7,9.5C7,8.7 7.7,8 8.5,8C9.3,8 10,8.7 10,9.5M12,14C13.75,14 15.29,14.72 16.19,15.81L14.77,17.23C14.32,16.5 13.25,16 12,16C10.75,16 9.68,16.5 9.23,17.23L7.81,15.81C8.71,14.72 10.25,14 12,14Z"
                      />
                    </svg>
                  </label>
                </div>
                <div>
                  <label
                    htmlFor="super-sad3"
                  >
                    <input
                      className="super_sad"
                      id="super-sad3"
                      name="recommended"
                      onChange={[Function]}
                      type="radio"
                      value={1}
                    />
                    <svg
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12,2C6.47,2 2,6.47 2,12C2,17.53 6.47,22 12,22A10,10 0 0,0 22,12C22,6.47 17.5,2 12,2M12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4A8,8 0 0,1 20,12A8,8 0 0,1 12,20M16.18,7.76L15.12,8.82L14.06,7.76L13,8.82L14.06,9.88L13,10.94L14.06,12L15.12,10.94L16.18,12L17.24,10.94L16.18,9.88L17.24,8.82L16.18,7.76M7.82,12L8.88,10.94L9.94,12L11,10.94L9.94,9.88L11,8.82L9.94,7.76L8.88,8.82L7.82,7.76L6.76,8.82L7.82,9.88L6.76,10.94L7.82,12M12,14C9.67,14 7.69,15.46 6.89,17.5H17.11C16.31,15.46 14.33,14 12,14Z"
                      />
                    </svg>
                  </label>
                </div>
              </div>
            </div>
          </div>
        </fieldset>
        <legend>
          comments
        </legend>
        <fieldset>
          <div>
            <div>
              <label
                htmlFor="question1"
              >
                website_question1
              </label>
              <textarea
                id="question1"
                name="question1"
                onChange={[Function]}
                value=""
              />
            </div>
          </div>
          <div>
            <div>
              <label
                htmlFor="question2"
              >
                website_question2
              </label>
              <textarea
                id="question2"
                name="question2"
                onChange={[Function]}
                value=""
              />
            </div>
          </div>
          <div>
            <div>
              <label
                htmlFor="question3"
              >
                website_question3
              </label>
              <textarea
                id="question3"
                name="question3"
                onChange={[Function]}
                value=""
              />
            </div>
          </div>
          <div>
            <div>
              <label>
                issues_screenshot
              </label>
              <div
                className="image-input-class"
              >
                <span
                  className="ant-upload-wrapper css-dev-only-do-not-override-42nv3w"
                >
                  <div
                    className="ant-upload ant-upload-select"
                  >
                    <span
                      className="ant-upload"
                      onClick={[Function]}
                      onDragOver={[Function]}
                      onDrop={[Function]}
                      onKeyDown={[Function]}
                    >
                      <input
                        accept=".jpg,.jpeg,.png,.gif"
                        disabled={false}
                        multiple={false}
                        onChange={[Function]}
                        onClick={[Function]}
                        style={
                          {
                            "display": "none",
                          }
                        }
                        type="file"
                      />
                      <button
                        className="ant-btn css-dev-only-do-not-override-42nv3w ant-btn-default ant-btn-lg"
                        disabled={false}
                        onClick={[Function]}
                        style={{}}
                        type="button"
                      >
                        <span
                          className="ant-btn-icon"
                          style={{}}
                        >
                          <span
                            aria-label="upload"
                            className="anticon anticon-upload"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="upload"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M400 317.7h73.9V656c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V317.7H624c6.7 0 10.4-7.7 6.3-12.9L518.3 163a8 8 0 00-12.6 0l-112 141.7c-4.1 5.3-.4 13 6.3 13zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
                              />
                            </svg>
                          </span>
                        </span>
                        <span>
                          choose_an_image
                        </span>
                      </button>
                    </span>
                  </div>
                </span>
              </div>
            </div>
          </div>
        </fieldset>
        <div>
          <button
            id="reset_button"
            onClick={[Function]}
            type="reset"
          >
            reset_btn
          </button>
          <br />
          <button
            data-testid="test_submit"
            id="submit_button"
            name="submit_button"
            type="submit"
          >
            submit_btn
          </button>
        </div>
      </form>
    </div>
  </div>,
]
`;
